### DISCUSSION TITLE
Kustomization wait and healthChecks with flagger's Canary object

### DISCUSSION THREAD

**Discussion by alexcastilio** [fluxcd/flux2#3385] [Q&A] [answered]
Hi, 

I tried to setup a Kustomization to reconcile a flagger's Canary CRD object into the cluster with wait=true and healthChecks but it does not get the Status of the Canary object as feedback about the reconciliation of the Kustomization. The object is reconciled into the cluster but there's no feedback on the Kustomization status, which is always reconciled, independently of the fail/success of the Canary object.

Could someone help me understand if there's a way of setting this feedback about Canary into the Kustomization? My end goal is to use the Kustomization to write the commit status into the repository according to the fail/success of the canary upgrade of an app.

Thanks!

**Comment by stefanprodan** [comment, marked as answer]
This is not supported, the reason is that a Flagger analysis usually takes a long time (tens of minutes to hours). Blocking Flux to wait for it to finish will starve all the threads in Flux and nothing will get reconciled while Canaries are running.

**Reply by alexcastilio** [reply]
Ok. Thank you!
